mantidproject/mantid
# Handle positive images (MaxEnt)

1. Code refactoring
2. Adds the ability to deal with positive images.

**To test:**
- Code review. The algorithm was refactored into different classes: 
  - `MaxentData` stores the information related to the signal (both in real and image space), which includes: the experimental data and errors, the reconstructed data and the reconstructed image. It also performs calculations related to the data, such as the chi-sq, search directions and quadratic coefficients, etc.
  - `MaxentEntropy` defines the necessary operations related to the entropy. This is an abstract base class from which two concrete classes are derived at the moment. The entropy formula to use depends on the type of image: if there is no restriction on the image (i.e. if it can take any real number, positive, negative or zero) we use the derived class `MaxentEntropyNegativeValues`. If the image is restricted to positive values only, `MaxentEntropyPositiveValues` is used.

These classes have been moved to a subdirectory of `Framework/Algorithms`, `MaxEnt/`. Feedback/comments/suggestions are very welcome :)
- Test the algorithm, you should be able to reproduce the examples in the documentation. There is a new section "Positive Images" explaining the new functionality.

Fixes #15151.

Link to [Release notes](http://www.mantidproject.org/index.php?title=Release_Notes_3_7_MuonAnalysis&diff=26376&oldid=26371).

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
